bin/test.sh

```diff
@@ -13,6 +13,17 @@ file_ext="php"
 function main {
   has_failures=0
 
+  if [ $# -ge 1 ] && [ -n "$1" ]; then
+    specific_exercise=$(find ./exercises/{practice,concept} -name $1 -type d | sort)
+    for exercise_dir in $specific_exercise; do
+        test "${exercise_dir}" "example"
+        if [[ $? -ne 0 ]]; then
+          has_failures=1
+        fi
+      done
+      return $has_failures
+  fi
+
```

Comment:
We need to handle `practice` and `concept` differently (`example` vs. `exemplar`). So I suggest modifing the existing loops by adding the `-name` conditionally:

```suggestion
  name_filter=()
  if [ -n "$1" ]; then
    name_filter=("-name" "$1")
  fi
```

and later on:

```
  all_practice_exercise_dirs=$(find ./exercises/practice -maxdepth 1 -mindepth 1 -type d "${name_filter[@]}" | sort)
```

and

```
  all_concept_exercise_dirs=$(find ./exercises/concept -maxdepth 1 -mindepth 1 -type d "${name_filter[@]}" | sort)
```